pkg/agent/proxy/proxier.go

```diff
@@ -978,9 +1013,24 @@ func NewProxier(
 		p.endpointsConfig = config.NewEndpointsConfig(informerFactory.Core().V1().Endpoints(), resyncPeriod)
 		p.endpointsConfig.RegisterEventHandler(p)
 	}
+	if p.multiclusterEnabled {
+		// We register event handler here since serviceCIDRIPv4 is only required
+		// when Antrea Multi-cluster is enabled.
+		fnRegisterServiceCIDREventHandler(p)
+	}
 	return p
 }
 
+func registerServiceCIDREventHandler(p *proxier) {
+	p.serviceCIDRProvider.AddEventHandler(func(serviceCIDRs []*net.IPNet) {
+		for _, cidr := range serviceCIDRs {
+			if cidr.IP.To4() != nil {
+				p.serviceCIDRIPv4 = cidr
```

Comment:
Do we need lock? @hongliangl

Reply:
After sync with hongliang, I refined this part without event handler, proxy will get Service CIDR before sync rules.